cime_config/testdefs/testmods_dirs/clm/FatesSetupParamBuild/shell_commands

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+# Make sure the environment is setup to run the FATES modify parameter file tool
+
+# Function to check for errors and abort
+check_error () {
+   # Arguments: error, message
+   if [ $# != 2 ]; then
+      echo "Wrong number of arguments to check_error"
+   fi
+   error=$1
+   msg=$2
+   if [ $error != 0 ]; then
+      echo "ERROR: Command failed"
+      echo $msg
+      exit 1
+   fi
+}
+
+# If under a casedirectory get a few variables for later use
+if [ -f xmlquery ]; then
+   SRCDIR=$(./xmlquery SRCROOT --value || echo "null")
+   check_error $? "Trouble getting SRCROOT from case"
+   DEBUG=0
+# otherwise if this is being run in the testmod directory for debugging
+else
+   echo "set SRCDIR assuming running in the testmod directory"
+   DEBUG=1
+   PWD=`pwd`
+   SRCDIR="$PWD/../../../../.."
+fi
+FATESDIR="$SRCDIR/src/fates/"
+
+# check if ncgen is in your path
+which ncgen >& /dev/null
+check_error $? "ncgen is NOT in your path"
+if [ $DEBUG ]; then
+   echo "ncgen was found"
+fi
+
+# check if conda is in your path
+which conda >& /dev/null
+error=$?
+if [ $error != 0 ]; then
+   echo "conda is NOT in your path and is used to get the python environment to run the FATES modify parameter file tool"
+   noconda=1
+else
+   if [ $DEBUG ]; then
+      echo "conda was found"
+   fi
+   noconda=0
+fi
+
+# Check that  the modify script exists and can be used
+
+MODIFY_FATES_PARAMFILE="$FATESDIR/tools/modify_fates_paramfile.py"
+if [ ! -f $MODIFY_FATES_PARAMFILE ]; then
+   echo "$MODIFY_FATES_PARAMFILE does NOT exist"
+   exit 1
+elif [ $DEBUG ]; then
+   echo "$MODIFY_FATES_PARAMFILE was found"
+fi
+
+$MODIFY_FATES_PARAMFILE --help >& /dev/null
+error=$?
+if [ $error != 0 ]; then
+   echo "$MODIFY_FATES_PARAMFILE can NOT be successfully run"
+   if [ $noconda == 1 ]; then
+      echo "Attempting to activate the conda ctsm_pylib environment"
+      conda activate ctsm_pylib
+      check_error $? "Trouble activating the conda ctsm_pylib environment"
+      if [ $DEBUG ]; then
+         echo "conda activate ctsm_pylib was successful"
+      fi
+   else
+      echo "Make sure your python environment can run $MODIFY_FATES_PARAMFILE"
+      echo "One way to do that is to add conda to your environment"
+      echo "    then activate the ctsm_pylib environemnt"
+      echo "    in some cases you may have to add conda activate ctsm_pylib in your startup files"
+      echo "    ctsm_pylib is created at the top level of CTSM using py_env_create"
+      exit 1
```

Comment:
Add `>&2` to send to `STDERR` instead of `STDOUT` (all above lines).

Reply:
Done